### PR TITLE
Add basic dualization comparison to simple examples

### PR DESCRIPTION
**Pull request by jd-foster**
Happy for this be a separate PR: produces this table at the end of `simple_examples`:
```
Model Name         #PrimalVars #Cons #DualVars #DualCons
------------------------------------------------------------
theta_problem           15      7       6       1
minimum_distortion      11      15      14      15
max_cut                 10      5       4       1
k_means_clustering      21      29      28      22
```

https://jump.dev/JuMP.jl/previews/PR3408/tutorials/conic/simple_examples/

### CODE REVIEW

File: docs/src/tutorials/conic/simple_examples.jl [hunk 1]
```diff
@@ -416,7 +427,40 @@ function example_robust_uncertainty_sets()
...
+        continue
+    end
+    println(
+        rpad(string(sym), 20),
+        "\t",
+        num_variables(model),
+        "\t",
+        num_constraints(model; count_variable_in_set_constraints = true),
+        "\t",
+        num_variables(Dualization.dualize(model)),
```
[the remark this answers is not in the document]
Comment (jd-foster) [reply docs/src/tutorials/conic/simple_examples.jl:456]: Sure

File: docs/src/tutorials/conic/simple_examples.jl [hunk 2]
```diff
@@ -416,7 +427,38 @@ function example_robust_uncertainty_sets()
...
+# It may be more efficient to solve the dual of a conic program rather than
+# the original formulation.
+# We can compare the numbers of variables and constraints in the primal formulations
+# given here and in the dual formulation provided by the [Dualization.jl](@ref) package:
+
+println("Model Name \t   #PrimalVars #Cons #DualVars #DualCons")
+println("------------------------------------------------------------")
+for (sym, model) in sdp_model_dict
+    ## Skip models with constraints types not yet supported:
+    if sym in [:correlation_problem, :robust_uncertainty_sets]
```
Comment (blegat) [inline docs/src/tutorials/conic/simple_examples.jl:446]: What's not supported exactly ?

Comment (jd-foster) [reply docs/src/tutorials/conic/simple_examples.jl:446]: For `correlation_problem`:
```
ERROR: Constraints of the Function MathOptInterface.VectorAffineFunction{Float64} in the set MathOptInterface.PositiveSemidefiniteConeSquare are not yet implemented.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] supported_constraints(con_types::Vector{Tuple{Type, Type}})
   @ Dualization ~/.julia/packages/Dualization/Xb4KB/src/supported.jl:15
```
For `robust_uncertainty_sets`:
```
ERROR: Constraints of the Function MathOptInterface.ScalarAffineFunction{Float64} in the set MathOptInterface.Interval{Float64} are not yet implemented.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] supported_constraints(con_types::Vector{Tuple{Type, Type}})
   @ Dualization ~/.julia/packages/Dualization/Xb4KB/src/supported.jl:15
```

**Comment by odow** [comment]
I'm not sure if we need this part. Maybe let's hold off and see if we get any comments on discourse.

**Comment by jd-foster** [comment]
That's fine, probably need to resolve the dualization of the two outstanding problems anyway before adding this.

**Comment by odow** [comment]
PSD cone issue is https://github.com/jump-dev/Dualization.jl/issues/73. I don't know if `Interval` has a nice dual set.

**Comment by blegat** [comment]
Dualization does not have to define dualization for all sets because for sets for which it's not defined, it's going to be bridged before the dualization layer.
Anyway, checking the number of variables and constraints with `dualize` wouldn't give what we want. What we should check is the number of variables and constraints after bridging

**Pull request closed without merging**